spec/services/ping_admins_spec.rb

```diff
@@ -0,0 +1,22 @@
+require "rails_helper"
+
+RSpec.describe PingAdmins do
+  let(:subject) { described_class }
+  let(:user) { create(:user) }
+
+  it "pings slack in the production environment" do
+    stub_production do
```

Comment:
I'm not comfortable with stubbing production, I would rather leave this piece of code without a test and later think of refactoring where we wouldn't need to check for the environment in the code.
@rhymes what do you think?

Reply:
I'm not either, it could lead to surprising behaviors if it leaks in other tests. I'd rather refactor the code or mock the Slack bot expliclty but I agree it shouldn't be part of this PR.

I believe we're leaning on `Rails.env.production?` in the code because we're not using settings or feature flags at all yet.

Reply:
Agree on the surprising behavior. On the env checks, there are several checks like this in the codebase, and I'm not a fan of them because they make testing harder. I've implemented [a fake aws client](https://github.com/thepracticaldev/dev.to/blob/master/config/initializers/aws_sdk.rb) while working on a similar problem. Maybe not the cleanest solution, but I was able to get the code under the test.

Reply:
@lightalloy I think "faking" can be a good approach for external services. I generally chave FakeStripe in all my projects. Should I create a new issue so that we can discuss it in details over there? I'd like to solve this!

Reply:
@tannakartikey mocking vs using a tool like VCR could depend on how much the upstream API changes and how complicate it is to stub it, because you might end up testing the mock and not the actual API (at least if you interact directly with the API, not through a client) but that's a different argument.

Anyhow if you want to open an issue about removing production checks please do so, I'm not sure they can be removed right away

Reply:
Yep, I think it's worth creating an issue.